ttm_module_publish.gradle

```diff
@@ -0,0 +1,66 @@
+enum BuildType {
+    DEVELOP("-SNAPSHOT"), BETA("-BETA"), RELEASE(""), FEATURE("-FEATURE")
+
+    BuildType(String classifier) { this.classifier = classifier }
+    private final String classifier
+
+    String toString() { return classifier }
+
+    static fromBranch(String branch) {
+        // We use git flow - http://nvie.com/posts/a-successful-git-branching-model/
+        // The defaults use release/, hotfix/, etc instead of release- and hotfix- mentioned above
+        if (branch.equals("master")) {
+            //Our only releases are on master
+            return RELEASE
+        } else if (branch.startsWith("release/") || branch.startsWith("hotfix/")) {
+            //Both releases and hotfixes are prereleases - a new version that will show up as a full release later
+            return BETA
+        } else if (branch.equals("develop")) {
+            return DEVELOP
+        } else {
+            return FEATURE
+        }
+    }
+}
+
+apply plugin: 'maven-publish'
```

Comment:
[The Gradle documentation](https://docs.gradle.org/current/userguide/plugins.html#sec:build_scripts_only) says that `plugins{}` blocks can't be in script plugins, but that should be changed in future Gradle versions. Are applied plugins different? Does this actually remove the requirement of a consumer to apply the plugin themselves?